sprokit/src/sprokit/pipeline/pipeline.cxx

```diff
@@ -61,8 +60,52 @@
  * \brief Implementation of the base class for \link sprokit::pipeline pipelines\endlink.
  */
 
-namespace sprokit
+namespace sprokit {
+
+namespace {
+
+/**
+ * @brief Greatest common denominator
+ *
+ * @param a First number
+ * @param b Second number
+ *
+ * @return GCD
+ */
+template <typename T>
+T
+gcd( T a, T b )
 {
+  for ( ; ; )
```

Comment:
Is this guaranteed to return?  I haven't thought about it enough to understand why or why not, but it looks like the type of code that could have a corner case that would lead to an infinite loop.   Is there any documentation on what this algorithm is called,  and how it works?  Maybe a link to a webpage or something?  I know boost have several different algorithms for this that it was selecting between.

Reply:
I believe it will always converge. After each iteration, the values of a and b are reduced.
I can use Euclid's algorithm from Knuth so that the alg has a reference.